CMakeLists.txt

```diff
@@ -271,68 +266,176 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      endif()
+    endif()
+  endif()
+
+  if(NOT RTMIDI_FOUND)
+    message("!! Using bundled RtMidi")
+    add_library(RtMidi STATIC "src/midi/external/rtmidi/RtMidi.cpp")
+    target_include_directories(RtMidi PUBLIC "src/midi/external/rtmidi")
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+      target_compile_definitions(RtMidi PUBLIC "__LINUX_ALSA__")
+      target_link_libraries(RtMidi PUBLIC "asound")
+    elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+      target_compile_definitions(RtMidi PUBLIC "__WINDOWS_MM__")
+      target_link_libraries(RtMidi PUBLIC "winmm")
+    elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      target_compile_definitions(RtMidi PUBLIC "__MACOSX_CORE__")
+      find_library(COREMIDI_LIBRARY "CoreMIDI")
+      target_link_libraries(RtMidi PUBLIC "${COREMIDI_LIBRARY}")
+    endif()
+    target_link_libraries(OPL3BankEditor PRIVATE RtMidi)
   endif()
+
   target_sources(OPL3BankEditor PRIVATE "src/midi/midi_rtmidi.cpp")
   target_compile_definitions(OPL3BankEditor PRIVATE "ENABLE_MIDI")
-  target_link_libraries(OPL3BankEditor PRIVATE RtMidi)
 endif()
 
 if(USE_RTAUDIO)
-  add_library(RtAudio STATIC "src/audio/external/rtaudio/RtAudio.cpp")
-  target_include_directories(RtAudio PUBLIC "src/audio/external/rtaudio")
-  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    target_compile_definitions(RtAudio PUBLIC "__LINUX_ALSA__")
-    target_link_libraries(RtAudio PUBLIC "asound")
-    if(JACK_FOUND)
-      target_compile_definitions(RtAudio PUBLIC "__UNIX_JACK__")
-      target_include_directories(RtAudio PUBLIC ${JACK_INCLUDE_DIRS})
-      link_directories(${JACK_LIBRARY_DIRS})
-      target_link_libraries(RtAudio PUBLIC ${JACK_LIBRARIES})
+  if(NOT USE_VENDORED_RTAUDIO)
+    include(CheckCXXSymbolExists)
+    if(PKG_CONFIG_FOUND)
+      pkg_check_modules(RTAUDIO rtaudio>=5.1)
+      if(RTAUDIO_FOUND)
+        message("!! Using system-installed RtAudio found by pkg-config")
+        target_compile_options(OPL3BankEditor PUBLIC ${RTAUDIO_CFLAGS})
+        target_include_directories(OPL3BankEditor PUBLIC ${RTAUDIO_INCLUDE_DIRS})
+        target_link_libraries(OPL3BankEditor PUBLIC ${RTAUDIO_LDFLAGS})
+      else()
+        set(RTAUDIO_FOUND FALSE)
+      endif()
+    endif()
+
+    if(NOT RTAUDIO_FOUND)
+      find_library(RTAUDIO_MANUAL_LIBRARY rtaudio)
+      message("!! RtAudio library: ${RTAUDIO_MANUAL_LIBRARY}")
+      find_path(RTAUDIO_MANUAL_INCLUDE_DIRS NAMES "RtAudio.h" PATH_SUFFIXES "include/rtaudio")
+      message("!! RtAudio header directory: ${RTAUDIO_MANUAL_INCLUDE_DIRS}")
+      set(CMAKE_REQUIRED_INCLUDES ${RTAUDIO_MANUAL_INCLUDE_DIRS})
+      set(CMAKE_REQUIRED_LIBRARIES ${RTAUDIO_MANUAL_LIBRARY})
+      check_cxx_symbol_exists(RtAudio::getCompiledApiName "RtAudio.h" RTAUDIO_MANUAL_SUFFICIENT)
+      if(NOT RTAUDIO_MANUAL_LIBRARY-NOTFOUND AND NOT RTAUDIO_MANUAL_INCLUDE_DIRS STREQUAL RTAUDIO_MANUAL_INCLUDE_DIRS-NOTFOUND AND RTAUDIO_MANUAL_SUFFICIENT)
```

Comment:
> `NOT RTAUDIO_MANUAL_LIBRARY-NOTFOUND`

Isn't a mistake? :thinking: 
Please put strings into quotes also as they confuse some things here.

Reply:
I think I misread the docs w/r/t `find_library`'s `<var>-NOTFOUND` :man_facepalming:, will fix.

Reply:
Usually goes:
```cmake
find_library(KURWA kurwa)

if(KURWA)
     message("Found!")
endif()

if(NOT KURWA)
     message("Kurwa mać!")
endif()
```

Reply:
Was quite the hot day when I hacked that together, should be better now. :sweat_smile:

Reply:
Now looks much better! :fox_face: :+1: